bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py

```diff
@@ -12,14 +12,56 @@
 
 import logging
 
+from functools import singledispatch
+from typing import Any, List, Union
+
 import ROOT
 
 from DistRDF.CppWorkflow import CppWorkflow
+from DistRDF.Node import Node
+from DistRDF.Operation import Action, InstantAction, Operation
 from DistRDF.PythonMergeables import SnapshotResult
 
+
 logger = logging.getLogger(__name__)
 
 
+@singledispatch
+def append_node_to_actions(operation: Operation, node: Node, actions: List[Node]) -> None:
+    """
+    Appends the input node to the list of action nodes, if the operation of the
+    node is an action.
+    """
+    pass
+
+
+@append_node_to_actions.register(Action)
+@append_node_to_actions.register(InstantAction)
+def _(operation: Union[Action, InstantAction], node: Node, actions: List[Node]) -> None:
+    actions.append(node)
+
+
+@singledispatch
+def append_node_to_results(operation: Operation, promise: Any, results: list) -> None:
```

Comment:
Do you strictly need to define the overload for `Operation`?

Reply:
I need to define a first generic function to wrap with the `singledispatch` decorator. The [docs](https://docs.python.org/3.7/library/functools.html#functools.singledispatch) say that by default, the singledispatch would decorate the first argument with type `object` to be as generic as possible. Since we know that this function in general expects an `Operation` as the first type, I just thought to write the type hint there. But this is not really an overload since it's not wrapped by the `register` decorator

Reply:
I see, first you wrap it into this singledispatch thing and then you can use its register method. Got it.